### PR TITLE
Made modal steper not for fullscreen

### PR DESCRIPTION
**Pull request by MickhaelSh**
Made modal stepper, not for fullscreen
becmeATutor

### CODE REVIEW

**Review by rostyslaw27 — APPROVED** [review]

**Review by LubomyrKashuba — APPROVED** [review]

**Review by High-Voltaged — APPROVED** [review]

File: src/pages/mentor-home/MentorHome.js [hunk 1]
```diff
@@ -4,15 +4,14 @@ import BecomeATutor from '~/containers/mentor-home-page/become-a-tutor/BecomeATu
 import { ModalContext } from '~/context/modal-context'
 
 const MentorHome = () => {
```
Comment (maarekua) [inline src/pages/mentor-home/MentorHome.js:6]: It should have margins from body on x-axis 32px on y-axis 24px.

**Review by maarekua — APPROVED** [review]

File: src/pages/mentor-home/MentorHome.js (hunk 1 above)
Comment (maarekua) [inline src/pages/mentor-home/MentorHome.js:6]: It should have margins from body on x-axis 32px on y-axis 24px.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/pages/mentor-home/MentorHome.js (hunk 1 above)
Comment (MickhaelSh) [reply src/pages/mentor-home/MentorHome.js:6]: changed, added paperProps to modal

File: src/components/popup-dialog/PopupDialog.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ import useConfirm from '~/hooks/use-confirm'
 import useBreakpoints from '~/hooks/use-breakpoints'
 import { style } from '~/components/popup-dialog/popup-dialog.style'
 
-const PopupDialog = ({ content, closeModal, isFullScreen, setFullScreen }) => {
+const PopupDialog = ({ content, closeModal, isFullScreen, paperProps, setPaperProps, setFullScreen }) => {
```
Comment (maarekua) [inline src/components/popup-dialog/PopupDialog.js:9]: paperProps = {}

File: src/context/modal-context.js [hunk 3]
```diff
@@ -6,20 +6,23 @@ const ModalContext = createContext()
 const ModalProvider = (props) => {
   const [modal, setModal] = useState()
   const [isFullScreen, setFullScreen] = useState(false)
+  const [paperProps, setPaperProps] = useState({})
```
Comment (maarekua) [inline src/context/modal-context.js:9]: remove

Comment (MickhaelSh) [reply src/context/modal-context.js:8]: can't remove this, it`s needed to set props to context

File: src/components/popup-dialog/PopupDialog.js (hunk 2 above)
Comment (MickhaelSh) [reply src/components/popup-dialog/PopupDialog.js]: it has to be state so we can set props to context

File: src/context/modal-context.js [hunk 4]
```diff
@@ -5,21 +5,19 @@ const ModalContext = createContext()
 
 const ModalProvider = (props) => {
   const [modal, setModal] = useState()
-  const [isFullScreen, setFullScreen] = useState(false)
+  const [paperProps, setPaperProps] = useState({})
 
   const closeModal = useCallback(() => {
     setModal()
   }, [setModal])
```
[the remark this answers is not in the document]
Comment (MickhaelSh) [reply src/context/modal-context.js:34]: done

File: src/context/modal-context.js [hunk 5]
```diff
@@ -5,30 +5,42 @@ const ModalContext = createContext()
...
+  const closeModalAfterDelay = useCallback(
+    (delay = 5000) => {
+      const timerId = setTimeout(closeModal, delay)
+      setTimer(timerId)
+    },
+    [closeModal]
+  )
+
+  const openModal = useCallback(
+    ({ component, paperProps, delayToClose }) => {
```
Comment (maarekua) [inline src/context/modal-context.js:27]: ({props}, delayBeforeClose = 0)

Comment (MickhaelSh) [reply src/context/modal-context.js]: done

**Review by maarekua — APPROVED** [review]

**Pull request merged**